internal/storage/disk/dirwatch.go

```diff
@@ -86,6 +89,10 @@ func (dw *dirWatch) handleEvents(ctx context.Context) {
 			dw.processEvent(evtInfo)
 		case <-ticker.C:
 			dw.triggerUpdate()
+			_ = stats.RecordWithTags(context.Background(), []tag.Mutator{
+				tag.Upsert(metrics.KeyStoreDriver, DriverName),
+				tag.Upsert(metrics.KeyStorePollStatus, "success"),
```

Comment:
It seems a bit strange that this is unconditionally successful? I think the error handling is different between disk and git - disk just logs warnings and continues to apply other changes, whereas git stops on the first error?

Reply:
Yes, I noticed that myself whilst working on this PR. I think the reason they are different is that with the disk store, someone could potentially be in the middle of an operation (like copying an existing file and modifying it) and it would be better to just ignore that and continue rather than throwing an error and stopping. Whereas, with git, we only look at committed changes. If something is wrong, then it will always be wrong until there's a new commit.

I do agree that it's not great to have different behaviours but I didn't want to change it for this PR without fully understanding the implications.

Reply:
I wonder if a more useful metric would be the number of errors encountered within `triggerUpdate`?